File: server/src/main/java/com/smartexpense/api/application/dto/response/ExpenseResponseDto.java
package com.smartexpense.api.application.dto.response;

import lombok.Builder;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
public class ExpenseResponseDto {
    private UUID id;
    private BigDecimal amount;
    private String category;
    private String photoUrl;
    private String caption;
    private LocalDateTime expenseDate;
    private LocalDateTime createdAt;
}

File: server/src/main/kotlin/com/smartexpense/api/application/dto/response/ExpenseResponseDto.kt
package com.smartexpense.api.application.dto.response

import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.UUID

data class ExpenseResponseDto(
    val id: UUID,
    val amount: BigDecimal,
    val category: String,
    val photoUrl: String,
    val caption: String?,
    val expenseDate: LocalDateTime
)
